src/Compilers/CSharp/Portable/CodeGen/EmitExpression.cs

```diff
@@ -3483,6 +3487,22 @@ private void EmitConditionalOperator(BoundConditionalOperator expr, bool used)
         {
             Debug.Assert(expr.ConstantValueOpt == null, "Constant value should have been emitted directly");
 
+            // Generate branchless IL for (b ? 1 : 0).
+            if (used && _ilEmitStyle != ILEmitStyle.Debug &&
+                (IsNumeric(expr.Type) || expr.Type.PrimitiveTypeCode == Cci.PrimitiveTypeCode.Boolean) &&
```

Comment:
>IsNumeric(expr.Type) [](http://example.com/codeflow?start=17&length=20)

Do we have tests for each of the numeric types?

Reply:
Yes, here:

https://github.com/dotnet/roslyn/blob/e9aca836524661d3911716eae6b7c37cb731cbf3/src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenConditionalOperatorTests.cs#L125-L137